### PR TITLE
Enable building with 1.13 (besides 1.12)

### PR DESCRIPTION
**Pull request by wk8**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Upgrading to go 1.13

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by msau42** [comment]
Is there any particular feature of go1.13 you need? Right now the go version is controller by release-tools/prow.sh, which we need to apply to all the k8s sidecars. cc @pohly

**Comment by wk8** [comment]
> Is there any particular feature of go1.13 you need? Right now the go version is controller by release-tools/prow.sh, which we need to apply to all the k8s sidecars. cc @pohly

@msau42 : None. This is mostly at KK's request, for ease of development.

That being said, this patch only accounts for some minor changes that 1.13 needs. Still runs just fine with 1.12.

**Comment by ddebroy** [comment]
/approve

May be the title of the PR can be changed to "enable building with 1.13 (besides 1.12)".

**Comment by wk8** [comment]
> May be the title of the PR can be changed to "enable building with 1.13 (besides 1.12)".

Great suggestion :)

File: Makefile [hunk 1]
```diff
@@ -22,7 +22,7 @@ compile: compile-client compile-server compile-csi-proxy-api-gen
 
 .PHONY: compile-client
 compile-client:
-	$(GO_ENV_VARS) go build ./client/...
+	cd client && $(GO_ENV_VARS) go build ./...
```
Comment (msau42) [inline Makefile]: Sorry this is a little unrelated but I missed the recent changes to the Makefile. Do we want to test compilation and run unit tests as part of the pull job? If so, we need a "make test" rule here.  By moving away from release-tools/build.make, we also miss some rules for verifying gofmt and vendor too

Comment (ddebroy) [reply Makefile]: Agree. @wk8 do you think the Makefile was csi-proxy can be reconciled with the make targets under https://github.com/kubernetes-csi/csi-proxy/blob/master/release-tools/build.make? I.e. the common items can be driven by build.make while the special requirements like protobuf gen, code gen, etc. can be driven by the top level Makefile.

I think ^ should be handled in a separate PR.

**Comment by ddebroy** [comment]
/lgtm

The makefile re-alignment will happen in a separate PR from @wk8

**Pull request merged**